server/channels/api4/user.go

```diff
@@ -2210,15 +2211,49 @@ func revokeAllSessionsAllUsers(c *Context, w http.ResponseWriter, r *http.Reques
 	ReturnStatusOK(w)
 }
 
-func attachDeviceId(c *Context, w http.ResponseWriter, r *http.Request) {
-	props := model.MapFromJSON(r.Body)
+func handleDeviceProps(c *Context, w http.ResponseWriter, r *http.Request) {
+	receivedProps := model.MapFromJSON(r.Body)
+	deviceId := receivedProps["device_id"]
 
-	deviceId := props["device_id"]
-	if deviceId == "" {
-		c.SetInvalidParam("device_id")
```

Comment:
As of today, we haven't implemented any validation on the device id apart of the device id being set (as you see here).

Since we are changing this endpoint for the device id to be optional, there is no more validation to be made, so there is no way to get an invalid parameter.

We could look into adding some more validation, specially for the prefixes that a device id is supposed to have, but 0/5 on whether that is really necessary.

Reply:
A device may not have the device id for many reasons. But we still want to know the version number, right? If we are reusing the same endpoint, we want to be able to send the version number without the device id.

And if we want to add more stuff to the props in the future, the same. Therefore we remove the requirement of the device id.